docs/source/WritingTests.rst

```diff
@@ -224,17 +224,22 @@ In test you can access those params by:
 Running multiple variants of tests
 ==================================
 
-In previous section we describe the params handling so let's have a look on
-how to produce them and execute your tests with different params.
-
-The variants system is pluggable so you might use custom plugins to
-produce and feed avocado with your params, but let's start with the
-plugin called "yaml_to_mux", which is shipped with avocado by default.
-It accepts ``yaml`` or even ``json`` files where using ordered dicts
-to create a tree-like structure and storing the non-dict variables as
-parameters and using custom tags to mark locations as multiplex domains.
-Let's use ``examples/tests/sleeptenmin.py.data/sleeptenmin.yaml`` file as
-an example:
+In the previous section we described how parameters are handled.  Now,
+let's have a look at how to produce them and execute your tests with
+different parameters.
+
+The variants subsystem is what allows the creation of multiple
+variations of parameters, and the execution of tests with those
+parameter variations.  This subsystem is pluggable, so you might use
+custom plugins to produce variants.  To keep things simple, let's
+use Avocado's primary implementation, called "yaml_to_mux".
+
+The "yaml_to_mux" plugin accepts YAML files.  Those will create a
+tree-like structure, store the variables as parameters and use custom
+tags to mark locations as "multiplex" domains.
+
+Let's use ``examples/tests/sleeptenmin.py.data/sleeptenmin.yaml`` file
+as an example:
```

Comment:
@ldoktor my opinion here is that this may *currently* make sense to you, but most probably to you alone.  I really think I'm not alone in *not knowing* about how to use JSON with the `yaml_to_mux` (among core users/developers)

This is possible *technology* wise, but to me, it makes no sense *product* wise, unless we spend the extra effort in documenting, giving examples, and testing it.

Even the command line option help message strictly mentions YAML:

```
yaml to mux options:
  -m [FILE [FILE ...]], --mux-yaml [FILE [FILE ...]]
                        Location of one or more Avocado multiplex (.yaml)
                        FILE(s) (order dependent)
```

As for the survey, let's see if some users/developers (@apahim, @adereis, @lmr) can answer the question:

**Do you know how to use the `yaml_to_mux` plugin with JSON files?**